### ISSUE TITLE
CI: Update codesigning actions for both Windows and macOS 

### ISSUE THREAD

**Issue by shinra-electric** [higan-emu/ares#1483]
**Describe the bug**

All Windows CI runs have the following error: 
```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: DanaBear/code-sign-action@98c79121b376beab8d6a9484f445089db4461bca.
```

I can see from the GitHub notes that a specific build is used rather than a rolling release i.e. DanaBear/code-sign-action@master for security reasons. 

I was intending to make a PR to update it to `@v4` but I don't know if this would be appropriate from a security standpoint. 

Also, wpilibsuite/import-signing-certificate  has the same issue for the macOS runs: 
```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: wpilibsuite/import-signing-certificate@2ac4f44d28045073d23153256efbb4c4b2d8aa22.
```

 wpilibsuite/import-signing-certificate doesn't seem to use version tags, but `c9203157730f284175e15984cab690700bcf088f` would probably be the one to use. Again, I don't know if this is appropriate as it's regarding security.

I can make a PR with these changes if it's considered the right thing to do.

**Comment by rasky** [comment]
@shinra-electric yes please, you can do a PR and we will review it. It seems the best thing to do

**Comment by shinra-electric** [comment]
I've made a PR for the macOS action. But I made a note that it is not updated regularly. From a security standpoint this might not be the best. I suggested maybe looking at alternatives such as [Apple Actions](https://github.com/Apple-Actions/import-codesign-certs) which is often updated. There are other options as well. 

I didn't update the Windows signing action, because I couldn't find DanaBear on Github. ~~(Well, I could, but none or their repos were related)~~. Unless I'm missing something, it might be a good idea to look for an alternative for the Windows action too. 

Edit: I found it [here](https://github.com/Dana-Prajea/code-sign-action) though it also seems not to be regularly updated. The last commit was 3 years ago. It was originally forked from [this](https://github.com/dlemstra/code-sign-action) which does seem to be updated regularly.

**Comment by jcm93** [comment]
Under the new build system our macOS signing is up to snuff; rather than use an action at all we let `xcodebuild` handle it as part of the actual build process, which is the recommended / canonical method.

The Windows signing action remains the same; it is outdated, should be looked at and likely replaced. I personally do not have a Windows certificate, don't know the details of our current one or why the current signing action might be necessary, and I have not done the requisite research into the best path forward on Windows after this one's expiry. So while things seem to be working fine at present, Windows signing in general should probably be on our radar.
